lib/galaxy/webapps/tool_shed/model/__init__.py

```diff
@@ -246,10 +246,12 @@ def get_type_class( self, app ):
         return app.repository_types_registry.get_class_by_label( self.type )
 
     def get_tool_dependencies( self, changeset_revision ):
+        repo = hg.repository( ui.ui(), self.repo_path( app ) )
```

Comment:
`app` is not defined in this context, should be a function parameter?

See https://jenkins.galaxyproject.org/job/docker-toolshed/4129/testReport/junit/shed_functional.functional.test_0540_get_all_metadata_from_api/TestGetAllMetadata/test_0025_verify_dependency_json/